bn-apps/memberships-management/membership-service-contracts-and-states/src/main/kotlin/com/r3/businessnetworks/membership/states/Membership.kt

```diff
@@ -59,16 +57,33 @@ open class MembershipContract : Contract {
             }
         }
 
-        when (command.value) {
-            is Commands.Request -> verifyRequest(tx, command, outputMembership)
-            is Commands.SelfIssue -> verifySelfIssue(tx, command, outputMembership)
-            is Commands.Suspend -> verifySuspend(tx, command, outputMembership, tx.inputsOfType<MembershipState<*>>().single())
-            is Commands.Activate -> verifyActivate(tx, command, outputMembership, tx.inputsOfType<MembershipState<*>>().single())
-            is Commands.Amend -> verifyAmend(tx, command, outputMembership, tx.inputsOfType<MembershipState<*>>().single())
-            else -> throw IllegalArgumentException("Unsupported command ${command.value}")
+        when (tx.commands.size) {
+            1 -> {
+                val command = tx.commands.requireSingleCommand<Commands>()
+
+                when (command.value) {
+                    is Commands.Request -> verifyRequest(tx, command, outputMembership)
+                    is Commands.Suspend -> verifySuspend(tx, command, outputMembership, tx.inputsOfType<MembershipState<*>>().single())
+                    is Commands.Activate -> verifyActivate(tx, command, outputMembership, tx.inputsOfType<MembershipState<*>>().single())
+                    is Commands.Amend -> verifyAmend(tx, command, outputMembership, tx.inputsOfType<MembershipState<*>>().single())
+                    else -> throw IllegalArgumentException("Unsupported command ${command.value}")
+                }
+            }
+            2 -> verifySelfIssue(tx, tx.commands, outputMembership) // BNO self activate
```

Comment:
What is the logic behind this? Why 2 commands would mean self issue? Can BNO not just self issue state and then activate it straight away? I.e. it would require 1 extra flow for example SelfIssueStateFlow that would invoke IssueStateFlow and ActivateStateFlow, and also some minimal changes to the contract code to allow BNO and Member to be the same Party.